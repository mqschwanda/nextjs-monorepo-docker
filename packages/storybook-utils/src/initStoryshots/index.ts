import type { StoryshotsOptions } from '@storybook/addon-storyshots/dist/api/StoryshotsOptions';
import initAddonStoryshots from '@storybook/addon-storyshots';

/**
 * Options for the `initStoryshots` function.
 */
export type InitStoryshotsOptions = StoryshotsOptions & {
  /**
   * If the snapshot tests fail when react logs a key error.
   *
   * Set this to `false` if stories avoid using the key because it would show up in the code example.
   *
   * @default true
   */
  failOnReactKeyError?: boolean
};

/**
 * A wrapper for the `@storybook/addon-storyshots` module
 *
 * See [addon storyshots docs](https://storybook.js.org/addons/@storybook/addon-storyshots) for more information.
 */
export default function initStoryshots({
  failOnReactKeyError = true,
  ...options
}: InitStoryshotsOptions) {
  if (failOnReactKeyError === false) {
    const consoleError = jest.spyOn(console, 'error');

    beforeAll(() => {
      consoleError.mockImplementation((...args) => {
        const [message] = args;
        if (message.includes('Each child in a list should have a unique "key" prop.')) {
          return;
        }

        console.error(...args); // eslint-disable-line no-console
      });
    });

    afterAll(() => {
      consoleError.mockRestore();
    });
  }

  initAddonStoryshots(options);
}
