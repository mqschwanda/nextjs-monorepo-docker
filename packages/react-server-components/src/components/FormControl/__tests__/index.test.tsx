import { render } from '@testing-library/react';
import path from 'path';
import initStoryshots from '@storybook/addon-storyshots';
import { InputText, FormControl } from '@';

initStoryshots({
  configPath: path.resolve(__dirname, '../../../../../storybook/.storybook'),
  storyKindRegex: /^@mqs\/react-server-components\/components\/FormControl$/,
});

describe('@mqs/react-server-components', () => {
  describe('components', () => {
    describe('<FormControl />', () => {
      it('renders', () => {
        const testId = FormControl.name;

        const { getByTestId, asFragment } = render((
          <FormControl
            className='test'
          >
            <InputText
              placeholder='Type here'
            />
          </FormControl>
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
