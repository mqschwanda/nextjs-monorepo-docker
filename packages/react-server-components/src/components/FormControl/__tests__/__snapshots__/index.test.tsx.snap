// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mqs/react-server-components components <FormControl /> renders 1`] = `
<DocumentFragment>
  <div
    class="form-control test"
    data-testid="FormControl"
  >
    <input
      class="input input-md"
      placeholder="Type here"
    />
  </div>
</DocumentFragment>
`;

exports[`Storyshots @mqs/react-server-components/components/FormControl Form Control Example 1`] = `
<div
  className="form-control"
>
  <label
    className="label"
  >
    Text Input:
  </label>
  <input
    className="input input-bordered input-md"
    placeholder="type here..."
    type="text"
  />
  <label
    className="label"
  >
    <span
      className="label-text-alt"
    >
      input description goes here
    </span>
  </label>
</div>
`;
